### ISSUE TITLE
take allocation/deallocation back out of inner time/grid loop

### ISSUE THREAD

**Issue by andywood** [NCAR/GMET#28]
This was removed once, and put back in, but currently a set of fixed arrays are allocated and deallocated 10s of millions of times, when they can be allocated and deallocated just once -- in estimate_forcing_regression.f90.

**Comment by anewman89** [comment]
Some of these arrays are not actually fixed in time because of the addition of the dynamic predictor options.  The predictor arrays may change by time step if badly behaved predictors (e.g. NWP precipitation = 0 everywhere) are dropped, then brought back the following time step.  Solutions could be 1) define enough static predictor arrays for the possible combinations (not very sustainable long-term), or 2) define them to their maximum dimension and then only pass the necessary predictors to the various regression subroutines, or some other option.

**Comment by andywood** [comment]
This is true, there are more options on the dimension, but the permutations are still relatively finite.  I think one of the two suggested options would work.
